### ISSUE TITLE
How to visualize the feature map in your paper? 

### ISSUE THREAD

**Issue by kravrolens** [KevinJ-Huang/ECLNet#1]
Could you please describe how to **visualize** the feature map in your paper? Thank you!

**Comment by thisisqiaoqiao** [comment]
Have you found a solution?

**Comment by huangkevinj** [comment]
Sorry for the late response, you can refer to the code below and have a try:


feature_save(x,'/home/jieh/Projects/CommonTrain/LocalFinetune/output/experiments/'+folder+'x/',name)



import os
import cv2
#
def feature_save(tensor,name,i):
    # tensor = torchvision.utils.make_grid(tensor.transpose(1,0))
    tensor = torch.mean(tensor,dim=1)
    inp = tensor.detach().cpu().numpy().transpose(1,2,0)
    inp = inp.squeeze(2)
    inp = (inp - np.min(inp)) / (np.max(inp) - np.min(inp))
    if not os.path.exists(name):
        os.makedirs(name)
    # for i in range(tensor.shape[1]):
    #     inp = tensor[:,i,:,:].detach().cpu().numpy().transpose(1,2,0)
    #     inp = np.clip(inp,0,1)
    # # inp = (inp-np.min(inp))/(np.max(inp)-np.min(inp))
    #
    #     cv2.imwrite(str(name)+'/'+str(i)+'.png',inp*255.0)
    inp = cv2.applyColorMap(np.uint8(inp * 255.0), cv2.COLORMAP_JET)
    cv2.imwrite(name + '/' + i, inp)
